doc/development.md

```diff
@@ -0,0 +1,117 @@
+# Development
+
+A Go 1.7+ environment is required. Docker and rkt (with SELinux set to Permissive) should be installed.
+
+## Binaries
+
+Develop the `update-operator` and `update-agent` apps locally.
+
+```
+make
+```
+
+Build  the `update-operator` and `update-agent` binaries in a build image with rkt.
+
+```
+make release-bin
+```
+
+## Container Image
+
+Build a container image.
+
+```
+make image
+```
+
+Push a development image to a personal image repository to test it on a cluster.
+
+```
+make docker-push IMAGE_REPO=quay.io/USERNAME/container-linux-update-operator
+```
+
+## Verificaton
+
+### Cluster
+
+Deploy a Container Linux Kubernetes cluster that satisfies the [requirements](README.md#requirements).
+
+* [QEMU/KVM](https://github.com/coreos/matchbox/tree/master/examples/terraform/bootkube-install)
+* [Tectonic](https://github.com/coreos/tectonic-installer)
+
+In particular, be sure to mask `locksmithd.service` on every Container Linux node.
+
+```
+sudo systemctl mask locksmithd.service --now
+```
+
+### Deploy
+
+Edit `example/update-operator.yaml` and `example/update-agent.yaml` to refer to the development image. Create the `update-operator` deployment and `update-agent` daemonset.
```

Comment:
typo: should be `examples` folder. Also, `examples/update-agent.yaml` doesn't exist in this branch, although I don't think that'll be a problem when it gets merged.